### ISSUE TITLE
encode and decode are no longer completely reversible

### ISSUE THREAD

**Issue by stocks29** [bitwalker/exprotobuf#60]
When relying on default fields decoding may produce different values than the structures created with `new()`.

I have tests which confirm that encode/decode are reversible which worked fine with 1.2.1, but now fails with 1.2.4. The reason being, that the decoded structure has default values (0 or int, false for boolean) whereas the unspecified values for newly created structures all have values of `nil`.

```elixir
  @decoded MyProtobuf.new()

  test "can encode and decode my protobuf" do
    encoded = @decoded
    |> MyProtobuf.encode
    |> Base.encode64

    decoded = encoded
    |> Base.decode64!
    |> MyProtobuf.decode

    assert decoded == @decoded
  end
```

```
  1) test can encode and decode my protobuf (MyProtobuf)
     test/my_protobuf_test.exs:44
     Assertion with == failed
     code:  decoded == @decoded
     left:  %MyProtobuf{timestamp: 0, active: false}
     right: %MyProtobuf{timestamp: nil, active: nil}
     stacktrace:
       test/my_protobuf_test.exs:53: (test)
```

**Comment by amatalai** [comment]
It's little disappointing when patch version update that should fix elixir warnings makes your application totally broken.

I created temporary fix 
`{:exprotobuf, github: "amatalai/exprotobuf", branch: "proto2-fix"}` 
by reverting #56
Note that I have no intention to maintain separate version and it may be removed from my github when no longer necessary.

**Comment by bitwalker** [comment]
I've fixed the handling of default values in the constructors, and released a new point release 1.2.5 with the fix.

@amatalai It's certainly something I try to avoid doing, but I no longer actively use protobufs in any projects, and thus this project is maintained by community effort - if you fix a bug, and even go so far as to fork the repo and fix it in your fork, why wouldn't you open a PR, or at least link directly to the patch? It's about the lowest effort possible, and indeed would take less time than it took for you to write your comment. Even though it wasn't the right fix for this issue, it would've at least allowed others to see your patch, and I may have even merged it temporarily until I could get at the real fix. In any case, hopefully this latest fix resolves your need to maintain a separate fork.

**Comment by amatalai** [comment]
@bitwalker I didn't create PR because it was workaround not fix and as I said in previous comment I have no intention to maintain fork.
